server/controllers/profile.js

```diff
@@ -0,0 +1,138 @@
+const Profile =  require('../models/Profile')
+const User = require('../models/User')
+const asyncHandler = require("express-async-handler");
+
+exports.createProfile = asyncHandler(async (req, res, next) => {
+
+const {firstName , lastName , description , gender , dateOfBirth , phone ,availability } = req.body
+  const {id} = req.user
+  const findUserId = await Profile.findOne({ user: id });
+  if(findUserId){
```

Comment:
Naming could be better. Just `profileExists`. `findUserId` is not very descriptive and in fact misleading since we're actually getting the profile.